### ISSUE TITLE
Symbol's function definition is void: global-flycheck-mode

### ISSUE THREAD

**Issue by ZelphirKaltstahl** [flycheck/flycheck#1220]
Ultimately I want to get some linter working for python code and possibly other languages later.

I just installed `flycheck` from MELPA and added:

    (global-flycheck-mode)

to my `~/.emacs` file.

What happens: When I start emacs I get the following error:

    Warning (initialization): An error occurred while loading `/home/xiaolong/.emacs':
    
    Symbol's function definition is void: global-flycheck-mode
    
    To ensure normal operation, you should investigate and remove the
    cause of the error in your initialization file.  Start Emacs with
    the `--debug-init' option to view a complete error backtrace.

What I expected: I expected there to be no error at all.

I tried to reproduce the issue in `emacs -Q` but the instructions how to do that are unclear. What I tried to do and the problems I had:

1. Started using `emacs -Q`.
2. Read the guide on how to reproduce with `emacs -Q`. It says I should follow the installation instruction of the package. The installation instruction says I should put the following into my init file:

    (require 'package)
    
    (add-to-list 'package-archives
                 '("MELPA Stable" . "https://stable.melpa.org/packages/") t)
    (package-initialize)

That does not make sense, because `-Q` starts emacs without configuration files. When I do `C-h v user-init-file RET` it also tells me the variable is `nil`. Also I am not using any So I put it into the `*scratch*` buffer instead and selected from the menu bar `Lisp-Interaction --> Evaluate and Print`. That added a `t` in my scratch buffer.

3. Then I went on following the guide. It says I should run `M-x package-install RET flycheck`. This is where a problem appears: The only completion for anything with `fly` in the package name is `flylisp`. No flycheck there.
4. The guide for bug reports says I should run `M-x flycheck-verify-setup`, which also does not exist. There is no such function known to my emacs.

My emacs version is: `GNU Emacs 24.5.1 (x86_64-pc-linux-gnu, GTK+ Version 3.18.9) of 2016-04-17 on lgw01-04, modified by Debian`

My flycheck version: (same as with all the other flycheck functions, the function `flycheck-version` is not known)

Operating System: Xubuntu 16.04.2 LTS

To me it seems like something is wrong with the installation of flycheck, so that it shows this behavior.

How can I get it to run?
What more information can I provide / do I need to provide?

**Comment by fmdkdd** [comment]
Thank you for the (detailed) report 👍 

I agree with you, it definitely looks like Flycheck is not loading for you.

I tried to reproduce the installation through MELPA stable and it worked for me.  When you used `List-Interaction --> Evaluate and Print` from the menu, it should only have evaluated the last expression, not the whole `*scratch*` buffer.  That is why `flycheck` was not listed by `package-install`.

That said, if you already have Flycheck installed, running `package-install` in `emacs -Q` will not suggest `flycheck` itself (since it's already there in `.emacs.d/elpa`).

So, I've `mv .emacs.d .emacs.d.bak` to test from a clean slate.  And in the `*scratch*` buffer of an `emacs -Q`, I did:
```
(require 'package)
(add-to-list 'package-archives '("melpa-stable" . "https://stable.melpa.org/packages/") t)
(package-initialize)
(package-refresh-contents)
(package-install 'flycheck)
(global-flycheck-mode)
```
I evaluated each line by putting my cursor inside each and hit `C-M-x` (or `Lisp-Interaction -> Eval Defun`).  After the last line, Flycheck was loaded.

That was for the reproduction.  If you have already installed Flycheck in your regular Emacs, there should be a directory named `flycheck-...` under `.emacs.d/elpa/`.  Check that your `.emacs` also contains a `(package-initialize)` line before `(global-flycheck-mode)`.

**Comment by ZelphirKaltstahl** [comment]
Just a quick update: I just tried to `M-x package-install RET flycheck RET` again in my normal emacs. I got:

    Contacting host: melpa.org:443
    url-insert-file-contents: https://melpa.org/packages/flycheck-20170212.1015.tar: Not found

So maybe it didn't install itself properly in the first place (I only saw some warnings when it was compiling and ignored them) and that's why it the function is unknown. While there were warnings the first time around, this second time it cannot find the archive.

I'll get around to the process you describe above soon.

**Comment by ZelphirKaltstahl** [comment]
Seems the problem was with the installation process itself.

I tried installing again, this time clicking the menu items bringing up the list of packages in melpa, searching for flycheck, selecting it by clicking on it and clicking the install button in the shown buffer.

That installed without any warnings and now I got linting going on in Python and also in my emacs init file. Seems it worked fine and that installing it the way I initially tried somehow failed.

**Comment by fmdkdd** [comment]
Glad it works for you.  Happy hacking :)
